File: sdk/python/examples/63e_run_monitoring.py
# Copyright (c) 2025 Agentspan
# Licensed under the MIT License. See LICENSE file in the project root for details.

"""Run Monitoring Agent — trigger the monitoring agent deployed by 63d.

Demonstrates:
    - Running a deployed agent by workflow name from a separate process
    - The deploy/serve/run separation in practice

Requirements:
    - Conductor server running
    - 63d_serve_from_package.py running in another terminal
    - AGENTSPAN_SERVER_URL=http://localhost:8080/api in .env or environment
"""

from agentspan.agents import AgentRuntime


if __name__ == "__main__":
    with AgentRuntime() as runtime:
        print("Running monitoring agent by name...")
        result = runtime.run("monitoring", "Is everything healthy? Run a full check.")
        print(f"Workflow ID: {result.execution_id}")
        print(f"Output: {result.output}")

File: sdk/typescript/examples/63e-run-monitoring.ts
/**
 * 63e - Run Monitoring Agent — use runtime.run() and print the result.
 *
 * Requirements:
 *   - Conductor server running
 *   - AGENTSPAN_SERVER_URL=http://localhost:6767/api as environment variable
 *   - AGENTSPAN_LLM_MODEL=openai/gpt-4o-mini as environment variable
 */

import { monitoringAgent } from './63d-serve-from-package.js';
import { AgentRuntime } from '../src/index.js';

// Only run when executed directly (not when imported for discovery)
if (process.argv[1]?.endsWith('63e-run-monitoring.ts') || process.argv[1]?.endsWith('63e-run-monitoring.js')) {
  const runtime = new AgentRuntime();
  try {
    const result = await runtime.run(monitoringAgent, 'Is everything healthy? Run a full check.');
    result.printResult();

    // Production pattern:
    // 1. Deploy once during CI/CD:
    // await runtime.deploy(monitoringAgent);
    // CLI alternative:
    // agentspan deploy --package sdk/typescript/examples
    //
    // 2. In a separate long-lived worker process:
    // await runtime.serve(monitoringAgent);
  } finally {
    await runtime.shutdown();
  }
}
